caas/kubernetes/provider/cloud.go

```diff
@@ -92,61 +91,97 @@ func newCloudCredentialFromKubeConfig(reader io.Reader, cloudParams KubeCloudPar
 }
 
 // UpdateKubeCloudWithStorage updates the passed Cloud with storage details retrieved from the clouds' cluster.
-func UpdateKubeCloudWithStorage(k8sCloud *cloud.Cloud, storageParams KubeCloudStorageParams) (string, error) {
-	fail := func(e error) (string, error) {
-		return "", e
-	}
+func UpdateKubeCloudWithStorage(k8sCloud *cloud.Cloud, storageParams KubeCloudStorageParams) (storageMsg string, err error) {
 
 	// Get the cluster metadata so we can see if there's suitable storage available.
-	clusterMetadata, err := storageParams.GetClusterMetadataFunc(storageParams.MetadataChecker)
+	clusterMetadata, err := storageParams.GetClusterMetadataFunc(storageParams)
+
+	defer func() {
```

Comment:
This should be at the top of the function.
Can we move the code here into a new helper method

Reply:
yeah, moved a separate func